### DISCUSSION TITLE
Possibility of alterning the current behavior

### DISCUSSION THREAD

**Discussion by cig0** [Ubiquitine/virtual-desktops-only-on-primary#2] [General]
Hi,

The script works as advertised; thanks!

Would it be possible also to allow the opposite behavior? Ideally, I'd like to move through the virtual desktops on the `unfocused` screen, whether built-in or external.

Thanks again.

**Comment by Ubiquitine** [comment]
I'm not sure I understand exactly what you're asking.

You can change the windows pinning to a different screen. The script doesn't really know which screen is the primary. By default it takes the first screen that workspace.screens list (which is the laptop's built-in monitor). You can change the "primary" screen in the script configuration to you external monitor without actually setting it as primary. Just set "Primary display index #" to 1 in scripts configuration. As a result you should have virtual desktop only on your external monitor.

Regardless of above, you can switch virtual desktops with keyboard shortcuts. Even when the active screen (where your cursor is) is the one that "doesn't have virtual desktop", you can still switch virtual desktops with keyboard shortcuts, e.g. "Ctrl+Meta+Left/Right".

If your question described a different behavior, please try to describe it with more details. Maybe providing steps with the description of each step.

**Comment by cig0** [comment]
Hey! Sorry for leaving you hanging. I've been super busy these last few days. I'll get back to you later today.
